ss_admin_console/main/default/classes/feeder/UnitOfWorkHelper.cls

```diff
@@ -0,0 +1,123 @@
+public inherited sharing class UnitOfWorkHelper {
+    private DataSupplier supplier;
+    private String parentSObjectType;
+    private fflib_SObjectUnitOfWork uow;
+
+    public UnitOfWorkHelper(fflib_SObjectUnitOfWork uow, DataSupplier supplier) {
+        this.supplier = supplier;
+        this.uow = uow;
+        if (!supplier.getMetadata().referenceFields.isEmpty()) {
+            this.parentSObjectType = String.valueOf(supplier.getMetadata().referenceFields[0].referenceField.getDescribe().getReferenceTo()[0]);
+        }
+    }
+
+
+    public fflib_SObjectUnitOfWork registerRecords(List<SObject> data, Integer referenceFieldsSize, FeedLogger logger) {
+        if (referenceFieldsSize > 0) {
+            uow = registerChildsToParent(data, logger);
+        } else {
+            uow = registerParents(data);
+        }
+        return uow;
+    }
+
+    @TestVisible
+    private fflib_SObjectUnitOfWork registerParents(List<SObject> data) {      
+        for (SObject item : data) {
+            if (item.Id != null) {
+                uow.registerDirty(item);
+            } else {
+                uow.registerNew(item);     
+            }
+        }
+        return uow;
+    }
+
+    @TestVisible
+    private fflib_SObjectUnitOfWork registerChildsToParent(List<SObject> data, FeedLogger logger) {     
+        List<SObject> existentParentRecords = getRecords(data);
+        if (existentParentRecords.isEmpty()) {
+            logger.error('NOTE: Please, insert ' + parentSObjectType + '\'s records first.');
+            return uow;
+        }
+        Map<Integer, SObject> existentMap = generateRecordsMap(existentParentRecords);
+        Map<Integer, SObject> sampleDataMap = generateSampleRecordsMap(data, parentSObjectType);
+        for (Integer item : sampleDataMap.keySet()) {
+            SObject sampleSObject = sampleDataMap.get(item);
+            if (existentMap.containsKey(item)) {
+                SObject existentSObject = existentMap.get(item);
+                if (existentSObject != null) { // parent record exists
+                    sampleSObject.putSObject(parentSObjectType, existentSObject);
+                    if (sampleSObject.Id == null) { 
+                        uow.registerNew(sampleSObject); 
+                    } else { 
+                        uow.registerDirty(sampleSObject); 
+                    } 
+                    uow.registerRelationship(sampleSObject, supplier.getMetadata().referenceFields[0].referenceField, sampleSObject.getSObject(parentSObjectType));   
```

Comment:
existentSObject == sampleSObject.getSObject(parentSObjectType) ,  right?
existentSObject already has an Id, right?
so you can simplify it as :
samepleObject.set(supplier.getMetadata().referenceFields[0].referenceField, parentSObjectType.getId());